scripts/terminalserver/misc.jl

```diff
@@ -0,0 +1,57 @@
+# https://github.com/JuliaDebug/Debugger.jl/blob/4cf99c662ab89da0fe7380c1e81461e2428e8b00/src/limitio.jl
+
+mutable struct LimitIO{IO_t <: IO} <: IO
+    io::IO_t
+    maxbytes::Int
+    n::Int
+end
+LimitIO(io::IO, maxbytes) = LimitIO(io, maxbytes, 0)
+
+struct LimitIOException <: Exception end
+
+function Base.write(io::LimitIO, v::UInt8)
+    io.n > io.maxbytes && throw(LimitIOException())
+    io.n += write(io.io, v)
+end
```

Comment:
It seems impossible to me, but do you think can we respect "length" of strings instead of its bytes ?
If not, I guess `length(ellipsis)` inside of `sprintlimited` is better to be `ncodeunits(ellipsis)` ?
And maybe it makes sense to make `strlimit` behave consistently (i.e. respect bytes rather than length, maybe, or mention the difference if we keep them different)

Reply:
fwiw, here is our `strlimit` doc:
```julia
"""
    strlimit(str::AbstractString, limit::Integer = 30, ellipsis::AbstractString = "…")

Chops off `str` so that its _length_ doesn't exceed `limit`. The excessive part
  will be replaced by `ellipsis`.

!!! note
    The length of returned string will _never_ exceed `limit`.
"""
```

Reply:
The docstring still applies since `length(x) <= ncodeunits(x)` for all `x`. So yes, we might cut off too much and even produce invalid UTF8, but that's a price I'm willing to pay ;)

Reply:
aha, yeah true.
Don't know we need to be strict on the difference, so I'm going to approve this.